Bottom: da1e70cd8fc10de520644a8dad2aa12fe55f82c4
Top:    474266e0219aa3d67b3054345c4b24d3e5e1ecdc
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-29 23:39:21 +0100

draw.c: Fold DrawLayerCommon() into DrawLayer()

DrawLayer only calls DrawLayerCommon, so two functions are redundant.

DrawLayerCommon had one caller which passed TRUE for its "clear_pins"
argument, so special case that test in the caller.


---

diff --git a/src/draw.c b/src/draw.c
index fb29826..e0da9a9 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1072,18 +1072,15 @@ text_callback (const BoxType * b, void *cl)
   return 1;
 }
 
-/* ---------------------------------------------------------------------------
- * draws one non-copper layer
- */
 void
-DrawLayerCommon (LayerTypePtr Layer, const BoxType * screen, bool clear_pins)
+DrawLayer (LayerTypePtr Layer, const BoxType * screen)
 {
   struct poly_info info = {screen, Layer};
 
   /* print the non-clearing polys */
   r_search (Layer->polygon_tree, screen, NULL, poly_callback, &info);
 
-  if (clear_pins && TEST_FLAG (CHECKPLANESFLAG, PCB))
+  if (TEST_FLAG (CHECKPLANESFLAG, PCB))
     return;
 
   /* draw all visible lines this layer */
@@ -1110,12 +1107,6 @@ DrawLayerCommon (LayerTypePtr Layer, const BoxType * screen, bool clear_pins)
     }
 }
 
-void
-DrawLayer (LayerTypePtr Layer, const BoxType * screen)
-{
-  DrawLayerCommon (Layer, screen, true);
-}
-
 /* ---------------------------------------------------------------------------
  * draws one layer group.  If the exporter is not a GUI,
  * also draws the pins / pads / vias in this layer group.
@@ -1137,7 +1128,7 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
           strcmp (Layer->Name, "route") == 0)
         rv = 0;
       if (layernum < max_copper_layer && Layer->On)
-        DrawLayerCommon (Layer, drawn_area, true);
+        DrawLayer (Layer, drawn_area);
     }
   if (n_entries > 1)
     rv = 1;
